src/ts/components/transaction/Fee.tsx

```diff
@@ -20,7 +34,15 @@ class Fee extends React.Component<IFeeProps, IFeeState> {
     throw new Error(t('apiError'))
   }
 
+  componentDidUpdate (prevProps) {
+    if (this.props.toAddress !== prevProps.toAddress || this.props.address !== prevProps.address) {
+      this.updateBalance()
+      this.props.handleFeeChange(this.state.fee)
+    }
+  }
+
   updateBalance = () => {
+    if (this.props.toAddress.length !== 32) return
     if (this.props.apiContext.apiReady) {
       this.setState({ ...this.state, tries: 1 })
```

Comment:
you shouldn't need to retry here. we only connect to api once when the extension is loaded. You can check the logic in PopupApp.
```
tryCreateApi () {
    if (!this.props.apiContext.apiReady) {
      this.setState({ ...this.state, tries: this.state.tries++ })
      const network = networks[this.props.settings.network]
      const provider = new WsProvider(network.rpcServer)
      this.props.createApi(provider)
      if (this.state.tries <= 5) {
        // try to connect in 3 seconds
        setTimeout(this.tryCreateApi, 3000)
      }
    }
  }
```

Reply:
Does this apply to `Balance` component as well?

Reply:
I think so